packages/core/realtime-js/src/RealtimeChannel.ts

```diff
@@ -308,7 +308,11 @@ export default class RealtimeChannel {
 
       this.joinPush
         .receive('ok', async ({ postgres_changes }: PostgresChangesFilters) => {
-          this.socket.setAuth()
+          // FIX for issue #1904: Don't overwrite manually-set tokens
+          // Only refresh if token wasn't manually set via setAuth(token)
+          if (!this.socket['_manuallySetToken']) {
```

Comment:
I do not really like accessing private members like that. Can we please add a getter?